osu.Game/Screens/Multi/Match/Components/MatchBeatmapPanel.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Overlays.Direct;
+using osu.Game.Rulesets;
+
+namespace osu.Game.Screens.Multi.Match.Components
+{
+    public class MatchBeatmapPanel : MultiplayerComposite
+    {
+        [Resolved]
+        private IAPIProvider api { get; set; }
+
+        [Resolved]
+        private RulesetStore rulesets { get; set; }
+
+        private GetBeatmapSetRequest request;
+        private DirectGridPanel panel;
+
+        public MatchBeatmapPanel()
+        {
+            AutoSizeAxes = Axes.Both;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            CurrentItem.BindValueChanged(item =>
+            {
+                request?.Cancel();
+
+                if (panel != null)
+                {
+                    panel.FadeOut(200);
+                    panel.Expire();
+                    panel = null;
+                }
+
+                var onlineId = item.NewValue?.Beatmap.OnlineBeatmapID ?? 0;
```

Comment:
Why zero on null? the only usage I'm seeing here is checking, you can do `if (onlineId != null)`

Reply:
We are using GetBeatmapSetRequest with onlineId as an argument, which can't be nullable there

Reply:
you can use `onlineId.Value`, looking at the source they use `if (onlineId != null) ...(onlineId.Value)`, i guess for code quality you should match them

Reply:
sounds legit

Reply:
the request shouldn't be *made* if null, see my review above.